gaphor/diagram/general/generalpropertypages.py

```diff
@@ -148,6 +149,13 @@ def open_file(self, filename):
                         )
                         self.subject.width = image.width
                         self.subject.height = image.height
+                        if self.subject.subject.name in [None, gettext("New Picture")]:
+                            self.subject.subject.name = "".join(
+                                chr
+                                if chr.isalnum() or (chr in [" ", "_", "-"])
+                                else "_"
+                                for chr in Path(filename).stem
+                            )
```

Comment:
I would move this to a function, so it can be tested more easily.

Reply:
done